src/GraphQL.Tests/Subscription/ShouldlyExtensions.cs

```diff
@@ -0,0 +1,63 @@
+#nullable enable
+
+using GraphQL.Subscription;
+using GraphQL.SystemTextJson;
+
+namespace GraphQL.Tests.Subscription;
+
+public static class ShouldlyExtensions
+{
+    private static readonly IGraphQLTextSerializer _serializer = new GraphQLSerializer();
+
+    public static void ShouldBeSimilarTo(this object? actual, object? expected, string? customMessage = null)
+    {
+        if (expected is string str)
+            expected = _serializer.Deserialize<Inputs>(str);
+        var expectedJson = _serializer.Serialize(expected);
+        var actualJson = _serializer.Serialize(actual);
+        actualJson.ShouldBe(expectedJson, customMessage);
+    }
+
+    public static ExecutionResult ShouldHaveResult(this SampleObserver? observer)
+    {
+        observer.ShouldNotBeNull();
+        observer.Events.ShouldNotBeNull();
+        observer.Events.TryDequeue(out var result).ShouldBeTrue("Observable sequence should have another result but did not");
+        return result;
+    }
+
+    public static void ShouldHaveNoMoreResults(this SampleObserver? observer)
+    {
+        observer.ShouldNotBeNull();
+        observer.Events.ShouldNotBeNull();
+        observer.Events.TryDequeue(out var result).ShouldBeFalse("Observable sequence should not have additional results but did");
+    }
+
+    public static ExecutionResult ShouldNotBeSuccessful(this ExecutionResult result)
```

Comment:
result.Executed may be both true and false for this test, right?

Reply:
Correct.  Granted, this is a bit specific to the subscription code, as `Data` may not be null for unsuccessful queries.

Reply:
These extensions might be very nice to put more globally in GraphQL.Tests.  In my own project, I spent a whole bunch of time figuring out how to make Shouldly play nicely and give the proper error messages and so on, which I did not do here.  Perhaps sometime I will copy some of that here.  

Just FYI, these are what some of my tests look like (particularly with Theory tests):

```cs
var ret = await RunQueryAsync(
    @"
mutation ($original: SettingInput!, $modified: SettingInput!) {
  setting {
    edit (original: $original, modified: $modified) {
        id
        name
    }
  }
}",
    new {
        original = new {
            id = "1",
            name = "Test1",
            value = "Value1",
        },
        modified = new {
            id = "1",
            name = "Test2",
            value = "Value2",
        }
    });
ret.ShouldBeSuccessful();
// and/or:
ret.ShouldMatchApproved();
// or:
var expected = new {
    setting = new {
        edit = new {
            id = 1,
            name = "Test2",
        }
    }
};
ret.ShouldHaveData(expected);
// similar for ShouldHaveError
```
JSON strings are also supported rather than anonymous objects.

Reply:
The biggest issue I had was that during serialization, properties were not always serialized in the same order (depends on the compiler).  So I customized the serializer to alphabetically sort all maps for the purposes of comparison.